src/firebase-app.ts

```diff
@@ -312,6 +311,10 @@ export class FirebaseApp {
     });
   }
 
+  public registerService(serviceName: string, service: FirebaseServiceInterface): void {
+    this.services_[serviceName] = service;
+  }
```

Comment:
The relationship before for `firebase-app` and the individual services was different, as `firebase-app` actually used the `this.services_` map to lazily retrieve the services. 

Now that modules are used it isn't needed anymore, and the tear-down process invoked in each function as part of the `delete()` doesn't work unless each service registers with the respective app. The disadvantage of this approach is that it's not very clean as in each service function the service itself needs to register with the app.